### PR TITLE
[Carbon Black Cloud] Fix 404 job id response

### PR DESCRIPTION
**Pull request by nmccullagh-r7**
### Description
Fix error handling for 404 responses from carbon_black_cloud

### CODE REVIEW

File: plugins/carbon_black_cloud/unit_test/test_monitor_alerts.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def test_http_exceptions(
 
         self.assertEqual(expected_status_code, status_code)
         self.assertEqual(num_logs, len(response))
-        self.assertFalse(has_more_pages)
+        self.assertEqual(has_more_pages, status_code == 200)
```
Comment (joneill-r7) [inline plugins/carbon_black_cloud/unit_test/test_monitor_alerts.py:297]: can this be updated to `assertTrue`? just a bit strange we're asserting the status code on line 295 and then again in here to assert has_more_pages?

**Review by joneill-r7 — APPROVED** [review]
lgtm - just one comment around the assertion in the unit test

**Review by ekelly-r7 — APPROVED** [review]

File: plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py [hunk 2]
```diff
@@ -392,9 +405,13 @@ def _update_state_in_404(self, status_code: int, state: Dict[str, str], alerts_s
                 )
                 # Only delete the observation ID and the time this was triggered
                 # But keep the hashes and timings in the state for the next job
+                status_code = 200
+                has_more_pages = True
```
[the remark this answers is not in the document]
Comment (nmccullagh-r7) [reply plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py:409]: To trigger a new job

File: plugins/carbon_black_cloud/unit_test/test_monitor_alerts.py (hunk 1 above)
Comment (nmccullagh-r7) [reply plugins/carbon_black_cloud/unit_test/test_monitor_alerts.py:297]: No this should only be true in the case status_code == 200

Comment (joneill-r7) [reply plugins/carbon_black_cloud/unit_test/test_monitor_alerts.py:297]: Oh I hadn't expanded the PR further to see we have a bunch of other scenarios being tested. Makes sense!

**Pull request merged**